pkg/mapfs/fs.go

```diff
@@ -50,6 +50,13 @@ func (m *FS) Filter(skippedFiles []string) (*FS, error) {
 	if len(skippedFiles) == 0 {
 		return m, nil
 	}
+	filter := func(path string, _ fs.DirEntry) (bool, error) {
+		return slices.Contains(skippedFiles, path), nil
+	}
+	return m.FilterFunc(filter)
+}
+
+func (m *FS) FilterFunc(fn func(path string, d fs.DirEntry) (bool, error)) (*FS, error) {
 	newFS := New()
```

Comment:
Need to update this line after #4094 gets merged.

```suggestion
	newFS := New(mapfs.WithUnderlyingPath(m.underlyingPath))
```